### PR TITLE
Add test case for TransformSupportServices

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #CHT

**Type**
Choose one: Testing

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt [hunk 1]
```diff
@@ -57,4 +63,81 @@ class TransformSupportServicesTest : RobolectricTest() {
...
+  }
+
+  @Test
+  fun `createType() should return Time when given time`() {
+    Assert.assertTrue(transformSupportServices.createType("", "time") is TimeType)
+  }
+
+  @Test
+  fun `createResource() should add resource into output when given Patient and atRootOfTransForm as True`() {
+    transformSupportServices.createResource("", Patient(), true)
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:120]: Can we assert that the output size is 0 before invoking `createResource`?

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt [hunk 2]
```diff
@@ -57,4 +63,81 @@ class TransformSupportServicesTest : RobolectricTest() {
...
+  }
+
+  @Test
+  fun `createResource() should not add resource into output when given Patient and atRootOfTransForm as False`() {
+    transformSupportServices.createResource("", Patient(), false)
+    Assert.assertEquals(transformSupportServices.outputs.size, 0)
+  }
+
+  @Test
+  fun `resolveReference() should throw FHIRException this is not supported yet when given url`() {
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:131]: Does it throw this exception when given all urls? or should we be more specific about the condition of the url that causes the exception?

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt [hunk 3]
```diff
@@ -57,4 +63,81 @@ class TransformSupportServicesTest : RobolectricTest() {
...
+    Assert.assertThrows("resolveReference is not supported yet", FHIRException::class.java) {
+      transformSupportServices.resolveReference("", "https://url.com")
+    }
+  }
+
+  @Test
+  fun `performSearch() should throw FHIRException this is not supported yet when given url`() {
```
Comment (ndegwamartin) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:138]: [same as above](https://github.com/opensrp/fhircore/pull/985/files#r791514679)

**Comment by ndegwamartin** [comment]
@FikriMilano to fully test this class please confirm that there are no missing branches in the [class here ](https://github.com/opensrp/fhircore/blob/09da370fc28df170d9a4ed45ffacd8a7b9ee77d6/android/engine/src/main/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServices.kt#L58)

**Comment by ndegwamartin** [comment]
@FikriMilano to fully test this method please confirm that there are no missing branches in the [method here ](https://github.com/opensrp/fhircore/blob/09da370fc28df170d9a4ed45ffacd8a7b9ee77d6/android/engine/src/main/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServices.kt#L58)

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt (hunk 1 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:120]: Sure

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt (hunk 2 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:131]: Yeah, it throws exception regardless input, it's simply not supported yet.
https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServices.kt#L82...L85

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt (hunk 3 above)
Comment (FikriMilano) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt:138]: Same here, it throws exception regardless input.
https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServices.kt#L87...L90

**Comment by ndegwamartin** [comment]
> https://app.codecov.io/gh/opensrp/fhircore/compare/985/changes

Good stuff @FikriMilano

**Review by pld — APPROVED** [review]
